enterprise_access/apps/bffs/tests/test_response_builders.py

```diff
@@ -0,0 +1,140 @@
+from enterprise_access.apps.bffs.serializers import (
+    EnterpriseCustomerUserSubsidiesSerializer,
+    SubscriptionLicenseStatusSerializer
+)
+from enterprise_access.apps.bffs.tests.utils import TestHandlerContextMixin
+
+
+@ddt.ddt
+class TestBaseResponseBuilder(TestHandlerContextMixin):
+
+    @mock.patch('enterprise_access.apps.bffs.context.HandlerContext')
+    def test_base_build_error(self, mock_handler_context):
+        mock_handler_context.return_value = self.mock_handler_context
+        base_response_builder = BaseResponseBuilder(mock_handler_context)
+        with self.assertRaises(NotImplementedError):
+            base_response_builder.build()
+
+    @ddt.data(
+        {
+            'errors': True,
+            'warnings': True,
+        },
+        {
+            'errors': True,
+            'warnings': False,
+        },
+        {
+            'errors': False,
+            'warnings': True,
+        },
+        {
+            'errors': False,
+            'warnings': False,
+        }
+    )
+    @mock.patch('enterprise_access.apps.bffs.context.HandlerContext')
+    @ddt.unpack
+    def test_add_errors_warnings_to_response(self, mock_handler_context, errors, warnings):
+        mock_handler_context.return_value = self.mock_handler_context
+        mock_handler_context_instance = mock_handler_context.return_value
+        base_response_builder = BaseResponseBuilder(mock_handler_context_instance)
+        sample_response = {
+            "some_existing_metadata": "should_still_exist"
+        }
+        expected_output = {
+            **sample_response,
+            'errors': [],
+            'warnings': [],
+        }
+
+        if errors:
+            mock_handler_context_instance.errors.append(self.mock_error)
+            expected_output['errors'] = [self.mock_error]
+        if warnings:
+            mock_handler_context_instance.warnings.append(self.mock_warning)
+            expected_output['warnings'] = [self.mock_warning]
+        response_data = base_response_builder.add_errors_warnings_to_response(sample_response)
+        self.assertEqual(response_data, expected_output)
+
+    # TODO Revisit this function in ENT-9633 to determine if 200 is ok for a nested errored response
+    @ddt.data(
+        {
+            'status_code': status.HTTP_400_BAD_REQUEST
+        },
+        {
+            'status_code': None
+        }
+    )
+    @mock.patch('enterprise_access.apps.bffs.context.HandlerContext')
+    @ddt.unpack
+    def test_get_status_code(self, mock_handler_context, status_code):
+        mock_handler_context.return_value = self.mock_handler_context
+        mock_handler_context_instance = mock_handler_context.return_value
+        base_response_builder = BaseResponseBuilder(mock_handler_context_instance)
+        expected_output = status_code if status_code else 200
+        if status_code:
+            mock_handler_context_instance.status_code = status_code
```

Comment:
nit/inform: just a heads up, there is not currently a `status_code` property in `HandlerContext` as only the private `_status_code` property is exposed.